client/src/main/scala/org/http4s/client/JavaNetClientBuilder.scala

```diff
@@ -130,17 +118,18 @@ sealed abstract class JavaNetClientBuilder[F[_]] private (
           })
           _ <- F.delay(conn.setInstanceFollowRedirects(false))
           _ <- F.delay(conn.setDoInput(true))
-          resp <- blocker.blockOn(blocking(fetchResponse(req, conn)))
+          // TODO: fix the blocking here
```

Comment:
We've got an `Async` constraint here now, so we can just call `blocking` yeah

Reply:
oh, the problem is that `fetchResponse` is an effect rather than a thunk. So either we wrap precisely the blocking action with `blocking`, or we have to use `evalOn` plus a separate blocking EC

Reply:
Ok, so I think this is the blocking code:
```scala
req.body
              .through(writeOutputStream(F.delay(conn.getOutputStream), false))
              .compile
              .drain
```
fs2 will actually shift to the internal blocking pool for `writeOutputStream`, so the question is whether that's enough, or do we actually want to run the entire `writeResponse` effect on a blocking pool

Reply:
`conn.getResponseCode` also blocks on a `java.io.InputStream` awaiting the response line.  I can't find it from the source, but I also assume that `conn.getHeaderFields` does as well.  The JDK8 client is synchronous, making it a bit of an outlier in our ecosystem.

I didn't realize fs2 had an internal blocking pool.  I guess we could create one as well for API simplicity, or pass one in for user control?

Reply:
`conn.getResponseCode` also blocks on a `java.io.InputStream` awaiting the response line.  I can't find it from the source, but I also assume that `conn.getHeaderFields` does as well.  The JDK8 client is synchronous, making it a bit of an outlier in our ecosystem.

I didn't realize fs2 had an internal blocking pool.  I guess we could create one as well for API simplicity, or pass one in for user control?

Reply:
ah I wasn't clear about this -- CE3 exposes an internal blocking pool that is accessed via the signature:
```
trait Sync[F[_]] {
  def blocking[A](thunk: => A): F[A]
}
```
Fs2 uses that for its various blocking ops

Reply:
Alright I've made those two calls `blocking`, but there are a couple of other ones that I'm concerned about that we should look into more, particularly this whole method:
```scala
private def writeBody(req: Request[F], conn: HttpURLConnection): F[Unit] =
    if (req.isChunked)
      F.delay(conn.setDoOutput(true)) *>
        F.delay(conn.setChunkedStreamingMode(4096)) *>
        req.body
          .through(writeOutputStream(F.delay(conn.getOutputStream), false))
          .compile
          .drain
    else
      req.contentLength match {
        case Some(len) if len >= 0L =>
          F.delay(conn.setDoOutput(true)) *>
            F.delay(conn.setFixedLengthStreamingMode(len)) *>
            req.body
              .through(writeOutputStream(F.delay(conn.getOutputStream), false))
              .compile
              .drain
        case _ =>
          F.delay(conn.setDoOutput(false))
      }
```
going to open a follow up issue to track. (I hope you don't mind me creating all these follow ups, I think once this is merged a lot of other work will be unblocked :) )

Reply:
No, I think it's great.  It's a chaotic time between the dotty branch and this branch and I'm afraid of dropping things.

As noted in #4066, I think `fs2` is taking care of the blocking calls here, buried in `writeOutputStream`.